card/mod/01_core/set/all/tracked_attributes.rb

```diff
@@ -7,16 +7,29 @@ def assign_attributes args={}
     Card.set_specific_attributes.each do |key|
       @set_specific[key] = args.delete(key) if args[key]
     end
+    new_type_id = extract_type_id! args
+    subcard_args = extract_subcard_args! args
 
-    if newtype = args.delete('type')
-      args['type_id'] = Card.fetch_id newtype
+    if new_type_id
+      args['type_id'] = new_type_id
     end
-    @subcards = extract_subcard_args! args
     reset_patterns
   end
   params = ActionController::Parameters.new(args)
   params.permit!
+
   super params
+
+  if args
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.